cmd/api/src/api/v2/agi.go

```diff
@@ -258,7 +259,29 @@ func (s Resources) UpdateAssetGroupSelectors(response http.ResponseWriter, reque
 		if result, err := s.DB.UpdateAssetGroupSelectors(*ctx.FromRequest(request), assetGroup, selectorSpecs, false); err != nil {
 			api.HandleDatabaseError(request, response, err)
 		} else {
-			// When T0 asset group selectors are modified we must trigger analysis
+			// Update system tags for the updated selectors
+			nodeProperties := graph.NewProperties()
+			if assetGroup.Tag == model.TierZeroAssetGroupTag {
+				nodeProperties.Set(common.SystemTags.String(), model.TierZeroAssetGroupTag)
+			} else {
+				nodeProperties.Set(common.SystemTags.String(), model.OwnedAssetGroupTag)
+			}
+
+			for _, selectorSpec := range selectorSpecs {
+				if err := s.Graph.WriteTransaction(request.Context(), func(tx graph.Transaction) error {
+					if err := tx.Nodes().Filterf(func() graph.Criteria {
+						return query.Equals(query.NodeProperty(common.ObjectID.String()), selectorSpec.EntityObjectID)
+					}).Update(nodeProperties); err != nil {
```

Comment:
We also want to not necessarily overwrite all existing tags. If a node is already a part of tier zero, and we are adding it to owned via a selector, we don't want to overwrite the existing tier zero tag.